wordpress-mobile/WordPress-Android
# Issue/16150 blogging prompt list of respondents

Fixes #16150

This PR replaces the mock list of prompt respondents with an existing piece of code used to display users who like a blog post.

List of additional:
- I extracted "train of face" code into a separate package (`org.wordpress.android.ui.avatars`) and renamed it to something a bit more generic.
- Added support for separate label color in trailing text label.
- Plurals support.
- Updated the code that determines the width of the text label to work with any container. We need to manually set the width of that label, since it is located in RecyclerView and given the chance it will extend horizontally, instead of wrapping.
- List of `BloggingPromptRespondent` is now included in blogging prompt model.
- `BloggingPromptsOnboardingDialogFragment` is using the new adapter with some dummy data. Not sure at the moment what we will actually show there.

PR looks a bit big, but most of the content in `org.wordpress.android.ui.avatars` package is pre-existing. For some reason some of those files are shown as new.

1 answer:
[![Image from Gyazo](https://i.gyazo.com/0368fb2ab1d8f51c260e5faebe323dde.png)](https://gyazo.com/0368fb2ab1d8f51c260e5faebe323dde)

5 answers:
[![Image from Gyazo](https://i.gyazo.com/a4d54e4bf253837cf4a3b30865ca7872.png)](https://gyazo.com/a4d54e4bf253837cf4a3b30865ca7872)

No Answers:
[![Image from Gyazo](https://i.gyazo.com/bacb4858333c0fe70add2a41591ab8fd.png)](https://gyazo.com/bacb4858333c0fe70add2a41591ab8fd)


To test:
- Confirm that the blogging prompt respondents list looks as expected in both My Site card and Feature Dialog.
- Confirm that the list of likers looks the same in Post Details.

## Regression Notes
1. Potential unintended areas of impact
- "Train of Likers" in post detail might looks off.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Visual confirmation and existing tests.

3. What automated tests I added (or what prevented me from doing so)
BP card builder test.

PR submission checklist:
